pkg/storegateway/gateway_prepare_shutdown_http_test.go

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package storegateway
+
+import (
+	"bytes"
+	"context"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/kv/consul"
+	"github.com/grafana/dskit/ring"
+	"github.com/grafana/dskit/services"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/pkg/util/shutdownmarker"
+
+	"github.com/grafana/mimir/pkg/storage/bucket/filesystem"
+	"github.com/grafana/mimir/pkg/util/test"
+)
+
+func createStoreGateway(t *testing.T, reg prometheus.Registerer) *StoreGateway {
+	gatewayCfg := mockGatewayConfig()
+	gatewayCfg.ShardingRing.UnregisterOnShutdown = false
+
+	storageDir := t.TempDir()
+	storageCfg := mockStorageConfig(t)
+	storageCfg.TSDB.Dir = storageDir
+
+	ringStore, closer := consul.NewInMemoryClient(ring.GetCodec(), log.NewNopLogger(), nil)
+	t.Cleanup(func() { assert.NoError(t, closer.Close()) })
+
+	bucket, err := filesystem.NewBucketClient(filesystem.Config{Directory: storageDir})
+	require.NoError(t, err)
+
+	g, err := newStoreGateway(gatewayCfg, storageCfg, bucket, ringStore, defaultLimitsOverrides(t), log.NewNopLogger(), reg, nil)
+	require.NoError(t, err)
+	return g
+}
+
+func TestGatewayPrepareShutdownHttp_PrepareShutdownHandler(t *testing.T) {
```

Comment:
[nit] Just mapping the actual receiver + function you're testing.

```suggestion
func TestStoreGateway_PrepareShutdownHandler(t *testing.T) {
```